signup/serializers.py

```diff
@@ -162,7 +162,7 @@ class UserSerializer(serializers.ModelSerializer):
 
     class Meta:
         model = get_user_model()
-        fields = ('username', 'email', 'full_name')
+        fields = ('username', 'email', 'full_name', 'first_name', 'last_name')
```

Comment:
Why adding `first_name`, `last_name` back? We tried really hard so far to replace them by `full_name`.

Reply:
I was replicating the functionality in the templates. I'll remove them.

Reply:
But how do we update `full_name` if it's a computed field which consists of `first_name` and `last_name`?

Reply:
Update `first_name` and `last_name` in the User model by executing:

    first_name, mid, last_name = full_name_natural_split(seriallizer.validated_data['full_name'])
    if mid:
      first_name = first_name + ' ' + mid
    user.first_name = first_name
    user.last_name = last_name
    user.save()

It looks convoluted, but this is because we are trying here to have a clean and easy to understand API while dealing with some less-than-optimal choices made by Django.